src/sentry/tasks/commit_context.py

```diff
@@ -186,27 +194,33 @@ def process_commit_context(
                         },
                     )
 
-            if not commit:
-                # None of the commits found from the integrations' contexts exist in sentry_commit.
-
-                # We couldn't find the commit in Sentry, so we will debounce the task for 1 day.
-                # TODO(nisanthan): We will not get the commit history for new customers, only the commits going forward from when they installed the source-code integration. We need a long-term fix.
-                cache.set(cache_key, True, timedelta(days=1).total_seconds())
-
-                metrics.incr(
-                    "sentry.tasks.process_commit_context.aborted",
-                    tags={
-                        "detail": "commit_sha_does_not_exist_in_sentry",
-                    },
+            if not commit and new_commit:
+                context = new_commit["context"]
+                # If none of the commits exist in sentry_commit, we add the first commit we found
+                commit_author, _ = CommitAuthor.objects.get_or_create(
+                    organization_id=project.organization_id,
+                    email=context.get("commitAuthorEmail"),
+                    defaults={"name": context.get("commitAuthorName")},
                 )
+                commit = Commit.objects.create(
```

Comment:
should this be get_or_create

Reply:
nvm i see new_commit

Reply:
we already try to get the commit on L171 - we could change that to a get_or_create, but I'm not sure if we prefer using an existing Commit object instead of creating a new one?